test/unit-test/framework/workflows/workflow_status.c

```diff
@@ -0,0 +1,25 @@
+#include <ui/workflow_stack.h>
+#include <workflow/status.h>
+
+typedef struct {
+    void (*callback)(void*);
+    void* callback_param;
+} data_t;
+
+void _mock_status_spin(workflow_t* self)
+{
+    data_t* data = (data_t*)self->data;
+    if (data->callback) {
+        data->callback(data->callback_param);
+    }
+    workflow_stack_stop_workflow();
+}
+
+workflow_t* mock_workflow_status(void (*callback)(void*), void* callback_param)
```

Comment:
why can't you use the real `workflow_status`?

Reply:
This is used by the `workflow_cancel` unit test, so the status workflow is mocked (it already was, but had to mock the async version as well!) to make it a "proper" _unit_ test and avoid testing stuff that doesn't belong to it.